Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -2681,4 +2685,49 @@ void WebProcessPool::updateWebProcessSuspensionDelayWithPacing(WeakHashSet<WebPr
 
 #endif // ENABLE(WEB_PROCESS_SUSPENSION_DELAY)
 
+#if ENABLE(CONTENT_EXTENSIONS)
+
+constexpr static Seconds resourceMonitorRuleListCheckInterval = 24_h;
```

Comment:
Since this might involve fetching a resource from the server, we might want to add some random duration to this refresh interval to try and avoid issues with thundering herds.

Reply:
I've started with that lazy implementation with expiration. But that doesn't cover the case that one and only tab is running for more than 24 hours. If no request happens, list won't be updated. Well, no request, no new activity so this shouldn't be big deal, but covering these in timer may be straightforward.

Adding randomness is reasonable, but the timer began based on the timing of user launch and we don't need to think about the server access, do we?